Sources/NIOSOCKS/Messages/AuthenticationMethod.swift

```diff
@@ -0,0 +1,35 @@
+//===----------------------------------------------------------------------===//
+//
+// This source file is part of the SwiftNIO open source project
+//
+// Copyright (c) 2021 Apple Inc. and the SwiftNIO project authors
+// Licensed under Apache License v2.0
+//
+// See LICENSE.txt for license information
+// See CONTRIBUTORS.txt for the list of SwiftNIO project authors
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+//===----------------------------------------------------------------------===//
+
+import NIO
+
+/// The SOCKS authentication method to use, defined in RFC 1928.
+public struct AuthenticationMethod: Hashable {
```

Comment:
I'm not sure this data structure is going to be right: how does it handle usernames and passwords?